core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -992,15 +991,22 @@ private void listStatusInternal(LockedInodePath currInodePath,
       }
       DescendantType nextDescendantType = (descendantType == DescendantType.ALL)
           ? DescendantType.ALL : DescendantType.NONE;
+
+      String [] parentChildPathComp = null;
+      if (!((InodeDirectory) inode).getChildren().isEmpty()) {
+        String [] parentPathComp = PathUtils.getPathComponents(currInodePath.getUri().getPath());
```

Comment:
NIT: I think `parentComponents` is a better name.

Reply:
done